src/Umbraco.Core/PropertyEditors/DataValueReferenceFactoryCollection.cs

```diff
@@ -4,64 +4,148 @@
 
 namespace Umbraco.Cms.Core.PropertyEditors;
 
+/// <summary>
+/// Provides a builder collection for <see cref="IDataValueReferenceFactory" /> items.
+/// </summary>
 public class DataValueReferenceFactoryCollection : BuilderCollectionBase<IDataValueReferenceFactory>
 {
+    // TODO: We could further reduce circular dependencies with PropertyEditorCollection by not having IDataValueReference implemented
+    // by property editors and instead just use the already built in IDataValueReferenceFactory and/or refactor that into a more normal collection
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DataValueReferenceFactoryCollection" /> class.
+    /// </summary>
+    /// <param name="items">The items.</param>
     public DataValueReferenceFactoryCollection(Func<IEnumerable<IDataValueReferenceFactory>> items)
         : base(items)
-    {
-    }
+    { }
 
-    // TODO: We could further reduce circular dependencies with PropertyEditorCollection by not having IDataValueReference implemented
-    // by property editors and instead just use the already built in IDataValueReferenceFactory and/or refactor that into a more normal collection
-    public IEnumerable<UmbracoEntityReference> GetAllReferences(
-        IPropertyCollection properties,
-        PropertyEditorCollection propertyEditors)
+    /// <summary>
+    /// Gets all unique references from the specified properties.
+    /// </summary>
+    /// <param name="properties">The properties.</param>
+    /// <param name="propertyEditors">The property editors.</param>
+    /// <returns>
+    /// The unique references from the specified properties.
+    /// </returns>
+    public ISet<UmbracoEntityReference> GetAllReferences(IPropertyCollection properties, PropertyEditorCollection propertyEditors)
     {
-        var trackedRelations = new HashSet<UmbracoEntityReference>();
+        var references = new HashSet<UmbracoEntityReference>();
 
-        foreach (IProperty p in properties)
+        foreach (IProperty property in properties)
         {
-            if (!propertyEditors.TryGet(p.PropertyType.PropertyEditorAlias, out IDataEditor? editor))
+            if (!propertyEditors.TryGet(property.PropertyType.PropertyEditorAlias, out IDataEditor? dataEditor))
             {
                 continue;
             }
 
-            // TODO: We will need to change this once we support tracking via variants/segments
-            // for now, we are tracking values from ALL variants
-            foreach (IPropertyValue propertyVal in p.Values)
+            // Only use edited value for now
+            references.UnionWith(GetReferences(dataEditor, property.Values.Select(x => x.EditedValue)));
```

Comment:
I've changed this to also include the published value in PR https://github.com/umbraco/Umbraco-CMS/pull/15596, because we should still show a warning when trying to delete a media item that is only referenced in the published version of a page!

The only 'downside' would be that you still see the relation when you don't have a reference to the item in the current draft anymore. But that seems like a minor thing compared to unknowingly deleting a media item that is still referenced on a published page 😄